Keen.NetStandard.Test/EventCacheTest.cs

```diff
@@ -0,0 +1,110 @@
+﻿using Keen.NetStandard.EventCache;
+using NUnit.Framework;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+
+
+namespace Keen.NetStandard.Test
+{
+    [TestFixture]
+    public class EventCacheTest : TestBase
+    {
+        static readonly object[] Providers = 
+        {
+            new object[] { new EventCacheMemory() }
+        };
+
+        [Test]
+        [TestCaseSource("Providers")]
+        public void AddEvent_Null_Throws(IEventCache cache)
+        {
+            Assert.ThrowsAsync<KeenException>(() => cache.Add(null));
+        }
+
+        [Test]
+        [TestCaseSource("Providers")]
+        public async Task AddEvent_ValidObject_Success(IEventCache cache)
+        {
+            await cache.Add(new CachedEvent("url", JObject.FromObject( new { Property = "Value" })));
```

Comment:
I don't like that `Add()` is awaitable and doesn't follow the naming guidelines...should be `AddAsync()` but that's not due to this change, of course. Maybe we just open an Issue and do it for next breaking change.